boring/src/ssl/error.rs

```diff
@@ -152,21 +153,42 @@ impl<S: fmt::Debug> StdError for HandshakeError<S> {
 
 impl<S: fmt::Debug> fmt::Display for HandshakeError<S> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match *self {
-            HandshakeError::SetupFailure(ref e) => write!(f, "stream setup failed: {}", e)?,
-            HandshakeError::Failure(ref s) => {
-                write!(f, "the handshake failed: {}", s.error())?;
-                let verify = s.ssl().verify_result();
-                if verify != X509VerifyResult::OK {
-                    write!(f, ": {}", verify)?;
-                }
+        let (prefix, s) = match *self {
+            HandshakeError::SetupFailure(ref e) => {
+                return write!(f, "TLS stream setup failed:\n\n{}", e)
```

Comment:
This is kind of indirection which can cause errors in the future. Let's get rid of this early return